### PR TITLE
Add examples to Enum.slice/2 docs

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

**Review by ericmj — APPROVED** [review]

File: lib/elixir/lib/enum.ex [hunk 1]
```diff
@@ -2880,6 +2884,11 @@ defmodule Enum do
       iex> Enum.slice([1, 2, 3, 4, 5], 0..-1//2)
       [1, 3, 5]
 
+  To get every third element up to the 12th element:
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex:2887]:
```suggestion
  To get every third element up to the 12th element:
```

If you want up to 12th, then you need 0..12 (or 0..11)?

Comment (josevalim) [reply lib/elixir/lib/enum.ex:2887]: That's a bug!

```
iex(2)> Enum.slice(1..20, 0..10//2)
[1, 3, 5, 7, 9, 11]
iex(3)> Enum.slice(Enum.to_list(1..20), 0..10//2)
[1, 3, 5, 7, 9, 11, 13, 15, 17, 19]
```

The list version is wrong. :)

Also:

```
iex(4)> String.slice("abcdefghijklmnop", 0..10//2)
"acegik"
```

Comment (josevalim) [reply lib/elixir/lib/enum.ex:2887]: Pushing a fix soon.

Comment (josevalim) [reply lib/elixir/lib/enum.ex:2887]: Fixed in master.

Comment (whatyouhide) [reply lib/elixir/lib/enum.ex]: @josevalim nice, pushed a fix to this PR, let me know how it looks.

File: lib/elixir/lib/enum.ex [hunk 2]
```diff
@@ -2844,7 +2844,7 @@ defmodule Enum do
 
   `index_range` must be a `Range`. Given an `enumerable`, it drops
   elements before `index_range.first` (zero-base), then it takes elements
-  until element `index_range.last` (inclusively).
+  until element `index_range.last * index_range.step` (inclusively).
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex:2847]: This is wrong now, no?

File: lib/elixir/lib/enum.ex [hunk 3]
```diff
@@ -2856,6 +2856,10 @@ defmodule Enum do
   `enumerable`, or this one is greater than the normalized `index_range.last`,
   then `[]` is returned.
 
+  If a step `n` (other than `1`) is used in `index_range`, then it takes
+  every `n`th element from `index_range.first` to `index_range.last * n`
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex:2860]: The multiplication here is also wrong, no?

**Comment by whatyouhide** [comment]
@josevalim yep I forgot to go through the docs again 🤦 One more review, and we should be good to go 😉

**Review by josevalim — APPROVED** [review]

**Pull request merged**